cobc/scanner.l

```diff
@@ -1328,11 +1380,47 @@ read_literal (const char mark, const char *type)
 			}
 			plex_buff[i] = (cob_u8_t)c;
 		}
+
+		if (ebcdic_flag != CB_SYM_EBCDIC_INACTIVE) {
+			/* While scanning symbolic EBCDIC character: */
+			if (c == mark) {
+				if (ebcdic_flag == CB_SYM_EBCDIC_AFTER_SEP) {
+					cb_warning (COBC_WARN_FILLER,
+						    _("extraneous separator at end "
+						      "of symbolic EBCDIC characters"));
+				}
+				ebcdic_flag = CB_SYM_EBCDIC_INACTIVE;
+			} else if ('0' <= c && c <= '9') {
+				plex_buff[i++] = scan_ebcdic_char (c);
+				ebcdic_flag = CB_SYM_EBCDIC_ACTIVE;
+			} else if (c != ' ' && c != ',') {
+				cb_error (_("invalid character '%c' in sequence "
+					    "of symbolic EBCDIC characters"), c);
+				ebcdic_flag = CB_SYM_EBCDIC_ACTIVE;
+			} else if (ebcdic_flag == CB_SYM_EBCDIC_AFTER_SEP) {
+				cb_warning (COBC_WARN_FILLER,
```

Comment:
Isn't that an error, both here and above?

Reply:
Gcos documentation is not very explicit, it says that separators MUST be used between symbolic characters, but it does not say if using several of them or extra at the end is an error. Maybe we should have a warning only in non-relaxed syntax ?

Reply:
That's a 100% vendor extension with big use for that and little use for any non-GCOS code - so it should match the original as best as possible; testing on GCOS with presumably invalid data is necessary.

Reply:
Ok, I will have to ask David to perform such tests...

Reply:
So, on GCOS, misplaced commas generate an error (duplicates, before first char, after last char), but the behavior is different for spaces: multiple spaces are allowed, and space after last char too. I will try to mimic this behavior in the next version.

Reply:
done

Reply:
I think the new version mimic this behavior:

* spaces are ok everywhere
* extra commas generate errors